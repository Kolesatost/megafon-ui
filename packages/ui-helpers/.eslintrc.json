{
  "extends": "../../.eslintrc.json",
  "overrides": [
    {
      "files": ["*.test.tsx"],
      "rules": {
        "react/no-multi-comp": "off"
      }
    }
  ]
}
