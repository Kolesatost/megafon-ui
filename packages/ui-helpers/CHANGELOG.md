# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.0.0-beta.2](https://github.com/MegafonWebLab/megafon-ui/compare/@megafon/ui-helpers@1.0.0-beta.1...@megafon/ui-helpers@1.0.0-beta.2) (2021-10-07)


### Bug Fixes

* add lint-staged to pre-commit hook ([83e2e13](https://github.com/MegafonWebLab/megafon-ui/commit/83e2e1304c92cd0125e80fa0c2879e28476499d9))
* run lint-staged only on pre-commit hook ([8570564](https://github.com/MegafonWebLab/megafon-ui/commit/857056423e738135f6d0866df8c0166ce9bd289e))





# [1.0.0-beta.1](https://github.com/MegafonWebLab/megafon-ui/compare/@megafon/ui-helpers@1.0.0-beta.0...@megafon/ui-helpers@1.0.0-beta.1) (2021-10-04)


### Bug Fixes

* **ui-helpers:** eslint and prettier fixes ([920b13d](https://github.com/MegafonWebLab/megafon-ui/commit/920b13d232aa107e974b742a1bc11c0e49d4d955))
* **ui-helpers:** fix export types ([0e9d170](https://github.com/MegafonWebLab/megafon-ui/commit/0e9d17090780eaa2f916c21ae90b5c7e2355ba71))
* **ui-icons:** add frontend-presets; fix eslint and prettier errors ([c1e0675](https://github.com/MegafonWebLab/megafon-ui/commit/c1e06758c2b6002a3011d7e98a745f1f25186714))





# 1.0.0-beta.0 (2021-07-13)


### Features

* **packages:** added @megafon/ui-helpers package ([811f67c](https://github.com/MegafonWebLab/megafon-ui/commit/811f67c26c87e2582e461c03223b12344513ecac))


### BREAKING CHANGES

* **packages:** utils from @megafon/ui-core removed, use @megafon/ui-helpers package instead
* **packages:** breakpoints from @megafon/ui-core removed, use @megafon/ui-helpers package instead
* **packages:** throttleTime removed from export in @megafon/ui-core package
