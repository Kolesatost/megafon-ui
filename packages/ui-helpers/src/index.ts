export { default as cnCreate } from './cnCreate';
export { default as convert } from './convert';
export { default as detectTouch } from './detectTouch';
export { default as filterDataAttrs } from './filterDataAttrs';
export { default as breakpoints } from './breakpoints';
export { default as applyTheme } from './applyTheme';

export type { ConvertTransformConfig } from './convert';
export type { IFilterDataAttrs } from './filterDataAttrs';
