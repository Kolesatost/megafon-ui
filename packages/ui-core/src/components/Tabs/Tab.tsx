/* eslint-disable react/no-unused-prop-types */
import * as React from 'react';
import { IFilterDataAttrs } from '@megafon/ui-helpers';
import PropTypes from 'prop-types';

export interface ITabProps extends IFilterDataAttrs {
    /** Заголовок таба */
    title?: string;
    /** Иконка таба */
    icon?: React.ReactNode;
    /** Ссылка */
    href?: string;
    /** Функция рендера компонента-обертки для заголовка и иконки */
    renderTabWrapper?: (tab: React.ReactNode) => React.ReactNode;
}

const Tab: React.FC<ITabProps> = ({ children }) => <>{children}</>;

Tab.propTypes = {
    title: PropTypes.string,
    icon: PropTypes.node,
    href: PropTypes.string,
    renderTabWrapper: PropTypes.func,
    dataAttrs: PropTypes.objectOf(PropTypes.string.isRequired),
};

export default Tab;
