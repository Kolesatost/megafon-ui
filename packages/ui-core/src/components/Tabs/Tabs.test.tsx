import * as React from 'react';
import { cnCreate } from '@megafon/ui-helpers';
import Balance from '@megafon/ui-icons/basic-24-balance_24.svg';
import { shallow } from 'enzyme';
import Tab from './Tab';
import Tabs, { TabSize, TabHAlign, TabColorTheme } from './Tabs';

const renderTabWrapper = (tab: React.ReactNode) => <div className="tab-wrapper">{tab}</div>;

const cn = cnCreate('.mfui-tabs');
describe('<Tabs />', () => {
    afterAll(() => jest.clearAllMocks());

    it('should render with default props', () => {
        const wrapper = shallow(
            <Tabs>
                <Tab>1</Tab>
                <Tab>2</Tab>
            </Tabs>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('should render with props', () => {
        const wrapper = shallow(
            <Tabs
                className="className"
                classes={{
                    root: 'rootClass',
                    innerIndents: 'innerIndentsClass',
                    tab: 'tabClass',
                    activeTab: 'activeTabClass',
                }}
                size={TabSize.SMALL}
                hAlign={TabHAlign.CENTER}
                tabColorTheme={TabColorTheme.GREEN}
                sticky
                defaultIndex={1}
                renderOnlyCurrentPanel
            >
                <Tab title="title 1" icon={<Balance />} href="www.test.com">
                    1
                </Tab>
                <Tab title="title 2" renderTabWrapper={renderTabWrapper}>
                    2
                </Tab>
            </Tabs>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('should render without panels', () => {
        const wrapper = shallow(
            <Tabs>
                <Tab title="title 1" icon={<Balance />} href="www.test.com" />
                <Tab title="title 2" renderTabWrapper={renderTabWrapper} />
            </Tabs>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('chould call onTabClick', () => {
        const mockHandleTabClick = jest.fn();
        const wrapper = shallow(
            <Tabs onTabClick={mockHandleTabClick}>
                <Tab>1</Tab>
                <Tab>2</Tab>
            </Tabs>,
        );

        wrapper.find(cn('tab-inner')).last().simulate('click');

        expect(mockHandleTabClick).toBeCalledWith(1);
    });
});
