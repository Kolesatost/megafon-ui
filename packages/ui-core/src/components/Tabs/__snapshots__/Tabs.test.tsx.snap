// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs /> should render with default props 1`] = `
<div
  className="mfui-tabs mfui-tabs_size_medium mfui-tabs_h-align_left mfui-tabs_tab-color_white mfui-tabs_indents"
>
  <div
    style={
      Object {
        "height": "auto",
      }
    }
  >
    <div
      className="mfui-tabs__swiper-wrapper"
      style={
        Object {
          "paddingLeft": 0,
          "paddingRight": 0,
        }
      }
    >
      <Swiper
        className="mfui-tabs__swiper mfui-tabs__swiper_beginning"
        initialSlide={0}
        onFromEdge={[Function]}
        onReachBeginning={[Function]}
        onReachEnd={[Function]}
        onSwiper={[Function]}
        simulateTouch={false}
        slidesPerView="auto"
      >
        <SwiperSlide
          className="mfui-tabs__slide"
          key=".0"
        >
          <div
            className="mfui-tabs__tab"
          >
            <div
              className="mfui-tabs__tab-inner mfui-tabs__tab-inner_current"
              onClick={[Function]}
            />
          </div>
        </SwiperSlide>
        <SwiperSlide
          className="mfui-tabs__slide"
          key=".1"
        >
          <div
            className="mfui-tabs__tab"
          >
            <div
              className="mfui-tabs__tab-inner"
              onClick={[Function]}
            />
          </div>
        </SwiperSlide>
        <div
          className="mfui-tabs__underline"
          slot="wrapper-start"
          style={
            Object {
              "transform": "translateX(0px)",
              "transitionProperty": "none",
              "width": "0px",
            }
          }
        />
        <test-file-stub
          className="mfui-tabs__arrow mfui-tabs__arrow_prev mfui-tabs__arrow_hide"
          onClick={[Function]}
        />
        <test-file-stub
          className="mfui-tabs__arrow mfui-tabs__arrow_next"
          onClick={[Function]}
        />
      </Swiper>
    </div>
  </div>
  <div
    className="mfui-tabs__panel mfui-tabs__panel_current"
    key=".0"
  >
    <Tab>
      1
    </Tab>
  </div>
  <div
    className="mfui-tabs__panel"
    key=".1"
  >
    <Tab>
      2
    </Tab>
  </div>
</div>
`;

exports[`<Tabs /> should render with props 1`] = `
<div
  className="mfui-tabs mfui-tabs_size_small mfui-tabs_h-align_center mfui-tabs_tab-color_green className rootClass"
>
  <div
    style={
      Object {
        "height": "auto",
      }
    }
  >
    <div
      className="mfui-tabs__swiper-wrapper"
      style={
        Object {
          "paddingLeft": 0,
          "paddingRight": 0,
        }
      }
    >
      <Swiper
        className="mfui-tabs__swiper mfui-tabs__swiper_beginning innerIndentsClass"
        initialSlide={1}
        onFromEdge={[Function]}
        onReachBeginning={[Function]}
        onReachEnd={[Function]}
        onSwiper={[Function]}
        simulateTouch={false}
        slidesPerView="auto"
      >
        <SwiperSlide
          className="mfui-tabs__slide"
          key=".0"
        >
          <div
            className="mfui-tabs__tab tabClass"
          >
            <a
              className="mfui-tabs__tab-inner"
              href="www.test.com"
              onClick={[Function]}
            >
              <div
                className="mfui-tabs__tab-icon"
              >
                <test-file-stub />
              </div>
              <div
                className="mfui-tabs__tab-title"
              >
                title 1
              </div>
            </a>
          </div>
        </SwiperSlide>
        <SwiperSlide
          className="mfui-tabs__slide"
          key=".1"
        >
          <div
            className="mfui-tabs__tab tabClass activeTabClass"
          >
            <div
              className="tab-wrapper"
            >
              <div
                className="mfui-tabs__tab-inner mfui-tabs__tab-inner_current"
                onClick={[Function]}
              >
                <div
                  className="mfui-tabs__tab-title"
                >
                  title 2
                </div>
              </div>
            </div>
          </div>
        </SwiperSlide>
        <div
          className="mfui-tabs__underline"
          slot="wrapper-start"
          style={
            Object {
              "transform": "translateX(0px)",
              "transitionProperty": "none",
              "width": "0px",
            }
          }
        />
        <test-file-stub
          className="mfui-tabs__arrow mfui-tabs__arrow_prev mfui-tabs__arrow_hide"
          onClick={[Function]}
        />
        <test-file-stub
          className="mfui-tabs__arrow mfui-tabs__arrow_next"
          onClick={[Function]}
        />
      </Swiper>
    </div>
  </div>
  <div
    className="mfui-tabs__panel"
    key=".0"
  >
    <Tab
      href="www.test.com"
      icon={<test-file-stub />}
      title="title 1"
    >
      1
    </Tab>
  </div>
  <div
    className="mfui-tabs__panel mfui-tabs__panel_current"
    key=".1"
  >
    <Tab
      renderTabWrapper={[Function]}
      title="title 2"
    >
      2
    </Tab>
  </div>
</div>
`;
